execute/README.md

```diff
@@ -0,0 +1,458 @@
+In practical terms, this means that for any columns that are in the group key, the values in that table will all be the same.
+
+The table will also contain columns that are not referenced by the group key.
+These columns are free to be any value as long as they are all the same type.
+
+For a table, the following invariants should hold:
+
+* All values inside a column that is part of the group key should have the same value.
+* That value should match the group key's value.
+* The values in a column should all be of the same type.
+
+The last of these invariants is **loosely** held.
+This means that while they should be true, individual transformations may not make an effort to enforce them.
+That means that if a transformation would have its performance affected by enforcing the invariant and the invariant does not affect the transformation's correctness, the transformation may not enforce it when producing the output.
+For an example, the `group()` transformation does not enforce the last invariant and relies on downstream transformations to enforce it.
+
+The set of rows contained within a table are processed in chunks.
+A table chunk is a subset of the rows within a table that are in-memory.
+
+The rows within a table are **ordered**.
+When table chunks are sent, the rows within the first are all ordered before the rows in the second.
+Transformations may choose to change the order, or they may choose to have the order be meaningful.
+For example, the `derivative()` function gives meaning to the order of the input rows while `sort()` will rearrange the order.
+
+A table chunk is composed of a list of arrow arrays, each array corresponding to a column of the table.
+
+### Arrow Arrays
+
+[Apache Arrow](https://arrow.apache.org/) is a language-independent columnar format.
+The flux engine utilizes the Arrow library to represent the columnar data inside of table chunks.
+
+### Execution Pipeline
+
+The **execution pipeline** is the set of nodes passing data from a source to the result.
+The execution pipeline is composed of **nodes**.
+A node with no inputs is a **source** which produces data for nodes that are after, or downstream, of itself.
+A node that takes one or more inputs and produces an output is a **transformation**.
+The final node in the pipeline is the **result** which holds the results of the pipeline.
+
+You can think of the execution pipeline in flux code.
+
+    A |> B |> C
+
+This would translate into three nodes where `A` is a source and `B` and `C` are transformations.
+Flux code does not correspond directly with one function being one node.
+Some functions will be combined into a single node while other functions may get rearranged into other nodes or split into multiple nodes.
+It is the responsibility of the planner to convert flux code into an execution pipeline.
+
+### Dataset
+
+A dataset is a connector between transformations.
+Each transformation is associated with a single dataset.
```

Comment:
If it's a connector between transformations, wouldn't it be associated with two datasets? Input and output?

Reply:
Hm, maybe there's a better way to phrase this. The dataset is a connector, but it's more strongly associated with one transformations. It's responsible for storing data and keeping track of where to send messages. I'll try to rewrite this part.